server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -207,9 +210,12 @@ public ClusterState applyFailedShards(final ClusterState clusterState, final Lis
                     failedNodeIds = Collections.emptySet();
                 }
                 String message = "failed shard on node [" + shardToFail.currentNodeId() + "]: " + failedShardEntry.getMessage();
+                boolean nodeIsRestarting = Optional.ofNullable(clusterState.metadata().nodeShutdowns().get(shardToFail.currentNodeId()))
+                    .map(shutdownInfo -> shutdownInfo.getType().equals(Type.RESTART))
+                    .orElse(false);
                 UnassignedInfo unassignedInfo = new UnassignedInfo(UnassignedInfo.Reason.ALLOCATION_FAILED, message,
                     failedShardEntry.getFailure(), failedAllocations + 1, currentNanoTime, System.currentTimeMillis(), false,
```

Comment:
This is another case where we calculate `nodeIsRestarting` without really needing it.

Reply:
Not anymore!